### PR TITLE
Move password reset code to AuthenticationSubsystem

### PR DESCRIPTION
**Pull request by battermann**
https://wearezeta.atlassian.net/browse/WPB-8890

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by fisx — COMMENTED** [review]
glanced through about half of the PR.  looks good!

**Review by pcapriotti — COMMENTED** [review]
LGTM

**Comment by fisx** [comment]
one more failure (flake?  unrelated?  serious?) -

```
      failure - invalid signature:                                                                   FAIL
        Exception: Error executing request: token invalid
        CallStack (from HasCallStack):
          error, called at test/integration/API/OAuth.hs:417:26 in main:API.OAuth
        CallStack (from HasCallStack):
          error, called at src/Bilge/Assert.hs:101:18 in bilge-0.22.0-6Cxcfrpv0gi69IBgCcDsiP:Bilge.Assert
          <!!, called at src/Bilge/Assert.hs:109:19 in bilge-0.22.0-6Cxcfrpv0gi69IBgCcDsiP:Bilge.Assert
          !!!, called at test/integration/API/OAuth.hs:419:82 in main:API.OAuth
        Use -p '(!/turn/&&!/user.auth.cookies.limit/)&&/failure - invalid signature/' to rerun this test only.
```

**Comment by fisx** [comment]
> one more failure (flake? unrelated? serious?) -
> 
> ```
>       failure - invalid signature:                                                                   FAIL
>         Exception: Error executing request: token invalid
>         CallStack (from HasCallStack):
>           error, called at test/integration/API/OAuth.hs:417:26 in main:API.OAuth
>         CallStack (from HasCallStack):
>           error, called at src/Bilge/Assert.hs:101:18 in bilge-0.22.0-6Cxcfrpv0gi69IBgCcDsiP:Bilge.Assert
>           <!!, called at src/Bilge/Assert.hs:109:19 in bilge-0.22.0-6Cxcfrpv0gi69IBgCcDsiP:Bilge.Assert
>           !!!, called at test/integration/API/OAuth.hs:419:82 in main:API.OAuth
>         Use -p '(!/turn/&&!/user.auth.cookies.limit/)&&/failure - invalid signature/' to rerun this test only.
> ```

flake!

**Comment by fisx** [comment]
integration tests run times on CI:

```
base line: 9:26:10-9:49:48 => ~23
this PR, last run: 13:54:41-14:15:31 => ~19
```

not sure how representative these figures are, but they sure don't provide cause for worry.  :-)

File: services/brig/src/Brig/CanonicalInterpreter.hs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
```
Comment (akshaymankar) [inline services/brig/src/Brig/CanonicalInterpreter.hs:2]:
```suggestion
{-# OPTIONS_GHC -Wno-redundant-constraints #-}

```

Why?

**Review by akshaymankar — APPROVED** [review]

**Review by fisx — APPROVED** [review]

File: services/brig/src/Brig/User/Auth/Cookie.hs [hunk 2]
```diff
@@ -94,7 +94,7 @@ newCookie uid cid typ label = do
             cookieSucc = Nothing,
             cookieValue = tok
           }
-  DB.insertCookie uid c Nothing
+  adhocSessionStoreInterpreter $ Store.insertCookie uid (toUnitCookie c) Nothing
```
[the remark this answers is not in the document]
Comment (akshaymankar) [reply services/brig/src/Brig/User/Auth/Cookie.hs:97]: It was that, but then writing the mock was a pain. I thought it made more sense to make it `()` so its clear that the value is not stored.

**Pull request merged**